From a14a61c954e512b21fc1be4a408e945f7f15df17 Mon Sep 17 00:00:00 2001
From: Patrick Mooney <pmooney@pfmooney.com>
Date: Tue, 26 Jun 2018 20:52:19 +0000
Subject: [PATCH] OS-7054 bhyve should halt on triple fault Reviewed by: Mike
 Gerdts <mike.gerdts@joyent.com> Reviewed by: Hans Rosenfeld
 <hans.rosenfeld@joyent.com> Approved by: Hans Rosenfeld
 <hans.rosenfeld@joyent.com>

---
 usr/src/uts/i86pc/io/vmm/vmm.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/usr/src/uts/i86pc/io/vmm/vmm.c b/usr/src/uts/i86pc/io/vmm/vmm.c
index 0f722facb0..a4b669ab81 100644
--- a/usr/src/uts/i86pc/io/vmm/vmm.c
+++ b/usr/src/uts/i86pc/io/vmm/vmm.c
@@ -2324,9 +2324,7 @@ nested_fault(struct vm *vm, int vcpuid, uint64_t info1, uint64_t info2,
 	if (type1 == VM_INTINFO_HWEXCEPTION && vector1 == IDT_DF) {
 		VCPU_CTR2(vm, vcpuid, "triple fault: info1(%#lx), info2(%#lx)",
 		    info1, info2);
-#ifdef	__FreeBSD__
 		vm_suspend(vm, VM_SUSPEND_TRIPLEFAULT);
-#endif
 		*retinfo = 0;
 		return (0);
 	}
-- 
2.21.0

